testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -6718,6 +7169,41 @@ private static void checkIf(SqlOperatorFixture f) {
             + "'SORT_ARRAY\\(<INTEGER ARRAY>, <INTEGER>\\)'\\."
             + " Supported form\\(s\\): 'SORT_ARRAY\\(<ARRAY>\\)'\n"
             + "'SORT_ARRAY\\(<ARRAY>, <BOOLEAN>\\)'", false);
+
```

Comment:
This code is highly repetitive. There should be a way to reduce repetitions and make sure that the exact same code is executed modulo the array constructor. The test should be inside of a loop that iterates twice: once for constructors, once for functions. 

Here is a possible way to implement this, not necessarily the best:

The query is generated by some string substitution process, e.g., a helper function "array_query" that is invoked like: `array_query(false, "sort_array(%A%0)", "2, null, 1")`

First argument is a `false` for arrays, and `true` for constructors, the second argument is a pattern, and the following arguments are all strings, with values to substitute in the pattern for `%0`, `%1`, etc. In this example `%A%0` is substituted with either `array(%0)` or `array[%0]`; afterwards `%0` is a positional argument that is substituted with the first argument.

If some tests do not work with constructors perhaps a second flag is needed to skip these, but I have proposed in the past to allow constructors for empty arrays. Hopefully we'll implement that as well, and then all the tests should pass in both cases.

Reply:
@mihaibudiu thanks mihai for reviewing, good comments, agree with you. I till try to optimize this duplication.